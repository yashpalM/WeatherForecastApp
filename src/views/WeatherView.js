import React, { useCallback, useEffect } from 'react';
import { View, Text } from 'react-native';
import * as weatherActions from '../store/actions/WeatherAction';
import { useDispatch, useSelector } from 'react-redux';
import NetworkInfo from '../utils/NetworkInfo';
import Weather from '../component/Weather';


const WeatherView = props => {
    const { route, navigation } = props;
    const { data, errorMessage, isLoading } = useSelector(state => state.weather);
    const dispatch = useDispatch();
    const { zipcode } = route.params;
    useEffect(() => {
        fetchData()
    }, [dispatch]);


    const fetchData = useCallback(async () => {
        const isConnected = await NetworkInfo.isNetworkAvailable();
        if (isConnected == null) {
            dispatch(weatherActions.searchByZipcode(zipcode));
        } else {
        dispatch(
            isConnected
            ? weatherActions.searchByZipcode(zipcode)
            : weatherActions.getDataFromLocalStorage(zipcode),
        );
        }
        }, [dispatch]);

    return (
        <View>
            <Weather
                data={data}
                errorMessage={errorMessage}
                isLoading={isLoading}
            />
        </View>
    );


};

export default WeatherView;