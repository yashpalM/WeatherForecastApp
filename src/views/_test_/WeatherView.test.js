import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import WeatherView from '../WeatherView';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';


let wrapper;
const mockStore = configureMockStore([thunk]);
const store = mockStore({
    weather: {
        data: {},
        errorMessage: '',
        isLoading: false
    }
});

wrapper = shallow(
    <Provider store={store}>
        <WeatherView route={{ getParam: jest.fn() }} />
    </Provider>
);

it('WeatherScreen renders correctly', () => {
    renderer.create(<Provider store={store}>
        <WeatherView route={{ getParam: jest.fn() }} />
    </Provider>).toJSON();
});

describe('WeatherScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        });
    });
});

// test('navigation options', () => {
//     const navProps = { navigation: { navigate: () => { } } };
//     const navigationOptions = WeatherScreen.navigationOptions(navProps);

//     expect(navigationOptions).toMatchSnapshot();
// });