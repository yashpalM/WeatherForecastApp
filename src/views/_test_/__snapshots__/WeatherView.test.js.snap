// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherView Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
